xlators/mgmt/glusterd/src/glusterd-op-sm.c

```diff
@@ -2830,6 +2832,151 @@ glusterd_set_shared_storage(dict_t *dict, char *key, char *value,
+            keylen = strlen(vme->key);
+            ret = dict_get_strn(priv->opts, vme->key, keylen, &deflt_val);
+            if (!deflt_val) {
+                // Check for the option in the volinfo dict
+                ret = dict_get_strn(volinfo->dict, vme->key, keylen,
+                                    &deflt_val);
+                if (ret == -ENOENT)
+                    // Check for glusterd default values
+                    deflt_val = glusterd_get_option_value(volinfo, vme->key);
+                if (!deflt_val) {
+                    if (vme->value) {
+                        deflt_val = vme->value;
+                    } else {
+                        // Check the option for tranlator default value
+                        ret = glusterd_get_value_for_vme_entry(vme, &deflt_val);
+                        if (ret)
+                            goto out;
+                        else if (ret == -2)
+                            continue;
+                    }
+                }
+            }
+        }
+    }
+
+    ret = glusterd_compare_bool_value(deflt_val, input_value);
+    deflt_val = NULL;
+
+out:
+    if (deflt_val)
+        GF_FREE(deflt_val);
+
+    return ret;
+}
+
+// Check the options set in the cluster/per-volume satisfy the dependency chain
+
+int
+glusterd_dependency_chain_check(glusterd_volinfo_t *volinfo, char *key,
+                                char *value, char **errstr)
+{
+    int ind = 0;
+    int j = 0;
+    int match_cnt = 0;
+    int ret = -1;
+    char *err_s = NULL;
+
+    // Going through each of the predicates in the prohibhited_clauses list and
+    // comparing the values for each of the options for a volume
+    while (prohibited_clauses[ind]) {
+        // Check if there are any predicates left in the list
+        if (!prohibited_clauses[ind][0].op)
+            break;
+
+        // Copy the predicate clause, which should be logged in case of failure
+        if (prohibited_clauses[ind][0].val == CLAUSE_LABEL) {
+            err_s = gf_strdup(prohibited_clauses[ind][0].op);
+        } else {
+            match_cnt = 0;
+            j = 0;
+            while (j >= 0) {
+                // Check the end of a list of options for single predicate
+                if (!prohibited_clauses[ind][j].op)
+                    break;
+
+                // If the option being set now, is present in the
+                // prohibhited_clauses list
+                if (key && strcmp(key, prohibited_clauses[ind][j].op) == 0) {
+                    ret = glusterd_compare_bool_value(
+                        value, prohibited_clauses[ind][j].val);
+                } else {
+                    // Else, compare the value of an option in the predicate
+                    // with the value set for the volume
+                    ret = glusterd_compare_option_value(
+                        volinfo, prohibited_clauses[ind][j].op,
+                        prohibited_clauses[ind][j].val);
+                }
+
```

Comment:
... having followed the above suggestion, now you can perform comparison at a single location by a mere equality check:

```
ret = (value_cmp == value_bool);
```